src/components/PopupWithForm.js

```diff
@@ -30,4 +32,12 @@ export default class PopupWithForm extends Popup {
       this.close();
```

Comment:
Needs correction: Сlose modal in the case of successful api response in `then` block in `index.js`, otherwise the form will close even if an error occurred and you will need to reopen the form and fill the values again.